crates/pipeline_manager/src/runner.rs

```diff
@@ -254,24 +246,75 @@ scrape_configs:
         &self,
         db: &ProjectDB,
         request: &NewPipelineRequest,
-        config_yaml: &str,
+        config_descr: &ConfigDescr,
         pipeline_id: PipelineId,
     ) -> AnyResult<Child> {
+        assert!(
+            config_descr.project_id.is_some(),
+            "pre-condition for start(): config.project_id is set"
+        );
+        let project_id = config_descr.project_id.unwrap();
+
+        // Assemble the final config by including all attached connectors.
+        async fn generate_attached_connector_config(
+            db: &ProjectDB,
+            config: &mut String,
+            ac: &AttachedConnector,
+        ) -> AnyResult<()> {
+            let ident = 4;
+            config.push_str(format!("{:ident$}{}:\n", "", ac.config.as_str()).as_str());
+            let ident = 8;
+            config.push_str(format!("{:ident$}stream: {}\n", "", ac.config.as_str()).as_str());
```

Comment:
Something's off here.  We're pushing the same string twice.

Reply:
it generates this part:

DEMOGRAPHICS:
   stream: DEMOGRAPHICS

shall we call the top-level thing something else than stream (e.g., connector-{id} or something?)

Reply:
- This will generate an invalid config when two input connectors connect to the same input stream of the circuit (because endpoint name (the first `DEMOGRAPHICS` above) must be unique).  Maybe we can just use uuid?
- The  `ac.config` stream should be renamed to `stream`.

> shall we call the top-level thing something else than stream (e.g., connector-{id} or something?)

Which one's the top-level thing?

Reply:
top-level-thing is not the "stream:", will rename it.